go/merge/three_way_test.go

```diff
@@ -12,62 +12,72 @@ import (
 )
 
 var (
-	aa1      = createMap("a1", "a-one", "a2", "a-two", "a3", "a-three", "a4", "a-four")
-	aa1a     = createMap("a1", "a-one", "a2", "a-two", "a3", "a-three-diff", "a4", "a-four", "a6", "a-six")
-	aa1b     = createMap("a1", "a-one", "a3", "a-three-diff", "a4", "a-four", "a5", "a-five")
-	aaMerged = createMap("a1", "a-one", "a3", "a-three-diff", "a4", "a-four", "a5", "a-five", "a6", "a-six")
-
-	mm1       = createMap()
-	mm1a      = createMap("k1", createMap(0, "a"))
-	mm1b      = createMap("k1", createMap(1, "b"))
-	mm1Merged = createMap("k1", createMap(0, "a", 1, "b"))
-
-	mm2       = createMap("k2", aa1, "k3", "k-three")
-	mm2a      = createMap("k1", createMap(0, "a"), "k2", aa1a, "k3", "k-three", "k4", "k-four")
-	mm2b      = createMap("k1", createMap(1, "b"), "k2", aa1b)
-	mm2Merged = createMap("k1", createMap(0, "a", 1, "b"), "k2", aaMerged, "k4", "k-four")
+	aa1      = []interface{}{"a1", "a-one", "a2", "a-two", "a3", "a-three", "a4", "a-four"}
+	aa1a     = []interface{}{"a1", "a-one", "a2", "a-two", "a3", "a-three-diff", "a4", "a-four", "a6", "a-six"}
+	aa1b     = []interface{}{"a1", "a-one", "a3", "a-three-diff", "a4", "a-four", "a5", "a-five"}
+	aaMerged = []interface{}{"a1", "a-one", "a3", "a-three-diff", "a4", "a-four", "a5", "a-five", "a6", "a-six"}
+
+	mm1       = []interface{}{}
+	mm1a      = []interface{}{"k1", []interface{}{"a", 0}}
+	mm1b      = []interface{}{"k1", []interface{}{"b", 1}}
+	mm1Merged = []interface{}{"k1", []interface{}{"a", 0, "b", 1}}
+
+	mm2       = []interface{}{"k2", aa1, "k3", "k-three"}
+	mm2a      = []interface{}{"k1", []interface{}{"a", 0}, "k2", aa1a, "k3", "k-three", "k4", "k-four"}
+	mm2b      = []interface{}{"k1", []interface{}{"b", 1}, "k2", aa1b}
+	mm2Merged = []interface{}{"k1", []interface{}{"a", 0, "b", 1}, "k2", aaMerged, "k4", "k-four"}
 )
 
 func tryThreeWayMerge(t *testing.T, a, b, p, expected types.Value, vs types.ValueReadWriter) {
 	merged, err := ThreeWay(a, b, p, vs)
 	if assert.NoError(t, err) {
-		assert.True(t, expected.Equals(merged))
+		assert.True(t, expected.Equals(merged), "%s != %s", types.EncodedValue(expected), types.EncodedValue(merged))
+	}
+}
+
+func tryThreeWayConflict(t *testing.T, a, b, p types.Value, contained string) {
+	_, err := ThreeWay(a, b, p, nil)
+	if assert.Error(t, err) {
+		assert.Contains(t, err.Error(), contained)
 	}
 }
 
 func TestThreeWayMergeMap_DoNothing(t *testing.T) {
-	tryThreeWayMerge(t, nil, nil, aa1, aa1, nil)
+	tryThreeWayMerge(t, nil, nil, cm(aa1), cm(aa1), nil)
```

Comment:
Could you have a test suite which provides the create-value function, instead of duplicating everything?